x-pack/plugins/integration_assistant/public/components/create_integration/create_integration_assistant/steps/data_stream_step/use_generation.tsx

```diff
@@ -0,0 +1,215 @@
+        }`;
+
+        reportGenerationComplete({
+          connector,
+          integrationSettings,
+          durationMs: Date.now() - generationStartedAt,
+          error: originalErrorMessage,
+        });
+
+        let errorMessage = originalErrorMessage;
+        const errorCode = e.body?.attributes?.errorCode as ErrorCode | undefined;
+        if (errorCode != null) {
+          errorMessage = i18n.ERROR_TRANSLATION[errorCode];
+        }
+        setError(errorMessage);
+      } finally {
+        setIsRequesting(false);
+      }
+    })();
+    return () => {
+      abortController.abort();
+    };
+  }, [
+    isRequesting,
+    onComplete,
+    setProgress,
+    connector,
+    http,
+    integrationSettings,
+    reportGenerationComplete,
+    notifications?.toasts,
+  ]);
+
+  const retry = useCallback(() => {
+    setError(null);
+    setIsRequesting(true);
+  }, []);
+
+  return { progress, error, retry };
+};
+
+async function runGeneration({
+  integrationSettings,
+  connector,
+  deps,
+  setProgress,
+}: RunGenerationProps): Promise<GenerationResults> {
+  let additionalProcessors: ESProcessorItem[] | undefined;
+  // logSamples may be modified to JSON format if they are in different formats
+  // Keeping originalLogSamples for running pipeline and generating docs
+  const originalLogSamples = integrationSettings.logSamples;
+  let logSamples = integrationSettings.logSamples;
+  let samplesFormat: SamplesFormat | undefined = integrationSettings.samplesFormat;
+
+  if (integrationSettings.samplesFormat === undefined) {
+    const analyzeLogsRequest: AnalyzeLogsRequestBody = {
+      packageName: integrationSettings.name ?? '',
+      dataStreamName: integrationSettings.dataStreamName ?? '',
+      logSamples: integrationSettings.logSamples ?? [],
+      connectorId: connector.id,
+      langSmithOptions: getLangSmithOptions(),
+    };
+
+    setProgress('analyzeLogs');
+    const analyzeLogsResult = await runAnalyzeLogsGraph(analyzeLogsRequest, deps);
+    if (deps.abortSignal.aborted) {
+      throw new Error('Aborted');
+    }
+    if (isEmpty(analyzeLogsResult?.results)) {
+      throw new Error('No results from Analyze Logs Graph');
+    }
+    logSamples = analyzeLogsResult.results.parsedSamples;
+    samplesFormat = analyzeLogsResult.results.samplesFormat;
+    additionalProcessors = analyzeLogsResult.additionalProcessors;
+  }
+
+  const ecsRequest: EcsMappingRequestBody = {
+    packageName: integrationSettings.name ?? '',
+    dataStreamName: integrationSettings.dataStreamName ?? '',
+    rawSamples: logSamples ?? [],
+    samplesFormat: samplesFormat ?? { name: 'json' },
+    additionalProcessors: additionalProcessors ?? [],
+    connectorId: connector.id,
+    langSmithOptions: getLangSmithOptions(),
+  };
+
+  setProgress('ecs');
+  const ecsGraphResult = await runEcsGraph(ecsRequest, deps);
+  if (deps.abortSignal.aborted) {
+    throw new Error('Aborted');
+  }
```

Comment:
@semd Is this a good approach... This will be caught in the first line of catch
```
      } catch (e) {
        if (abortController.signal.aborted) return;
```

Reply:
I am not sure if that's necessary, probably an error will be thrown at the point the abort is triggered.

Reply:
Removed and tested , abort throws an error and is handled in catch. Removed these in https://github.com/elastic/kibana/pull/194150/commits/93eb9d54bff48530f018b7dd919273177b844879